### ISSUE TITLE
tornado RuntimeError: finish() called twice

### ISSUE THREAD

**Issue by nimengliusha** [jupyter-incubator/nb2kg#47]
When i try to connect remote kernel gateway through nb2kg, the kernel is not connected

Here is the notebook log:
![image](https://user-images.githubusercontent.com/31717221/84119191-9b78e900-aa66-11ea-8824-1eb83a830ca8.png)

Any help or advice would be appreciated.

**Comment by nimengliusha** [comment]
Notebook v5.7.8,  jupyter_kernel_gateway v2.4.1, nb2kg v0.7.0

KernelGateway log:

![image](https://user-images.githubusercontent.com/31717221/84127549-024fcf80-aa72-11ea-8727-620cd7965050.png)

Ipynb view：

![image](https://user-images.githubusercontent.com/31717221/84127682-27dcd900-aa72-11ea-900f-cfcd2eb7ed91.png)

**Comment by kevin-bates** [comment]
Hmm - I haven't seen this, but also haven't used NB2KG in a while.

Are you in a position to try this on Notebook 6.x?  Starting in 6.0, NB2KG is built into Notebook and can be "configured" simply by running:

`jupyter notebook --gateway-url=${KG_URL}`

You'll want to disable the NB2KG extension and remove the config settings that substitute the kernel manager, etc.

**Comment by nimengliusha** [comment]
i am using Notebook v5,7,8

**Comment by kevin-bates** [comment]
Yes - understood.  I was hoping you could try with NB 6 as a datapoint.

**Comment by kevin-bates** [comment]
Hmm - I'm not able to reproduce this.  I tried NB 5.7.8 with NB2KG 0.7.0 and hit both JKG (2.4.1) and JEG (master branch) servers, created python kernels, used them, and shutdown.

My launch command was as documented on the README (except I added `--debug`):
```
export KG_URL=http://127.0.0.1:8889
jupyter notebook \
  --NotebookApp.session_manager_class=nb2kg.managers.SessionManager \
  --NotebookApp.kernel_manager_class=nb2kg.managers.RemoteKernelManager \
  --NotebookApp.kernel_spec_manager_class=nb2kg.managers.RemoteKernelSpecManager --debug
```

Here are the versions used in the JKG case - all within the same env.
[nb578_nb2kg070_jkg241_requirements.txt](https://github.com/jupyter/nb2kg/files/4752987/nb578_nb2kg070_jkg241_requirements.txt)

I suspect you might be using an older version of tornado than I am.
